internal/pkg/manifest/validate.go

```diff
@@ -1498,6 +1498,15 @@ func (s SidecarConfig) validate() error {
 	}
 	return s.ImageOverride.validate()
 }
+func (s SidecarConfig) validateSideCarImageFields() error {
+	if s.Image.Basic == nil && s.Image.Advanced.Location == nil && s.Image.Advanced.Build.isEmpty() {
```

Comment:
this can be simplified:
```suggestion
	if s.IsZero() {
```